### PR TITLE
Create Poll fixes

### PR DESCRIPTION
**Pull request by Jordie0608**
Adds cancel button to input prompts.
Fixes date validating not evaluating time of day.
Modified port of logging from Paradise.
Polls questions that are created without an option are auto-deleted, unless they are text replies which don't require options, as otherwise the poll would appear to users but be blank.
Replaces magic strings with defines.

### CODE REVIEW

**Comment by Jordie0608** [comment]
@Markolie This PR has been appended since https://github.com/ParadiseSS13/Paradise/pull/2284 was merged if  you want to port the new changes as well.

File: code/modules/admin/create_poll.dm [hunk 1]
```diff
@@ -5,6 +5,34 @@
 	if(!dbcon.IsConnected())
 		src << "<span class='danger'>Failed to establish database connection.</span>"
 		return
+	var/returned = create_poll_function()
+	if(returned)
+		var/DBQuery/query_check_option = dbcon.NewQuery("SELECT id FROM [format_table_name("poll_option")] WHERE pollid = [returned]")
```
Comment (duncathan) [inline code/modules/admin/create_poll.dm]: [You should do the opposite of what this song suggests](https://www.youtube.com/watch?v=_R4qb6_RPUc)

**Comment by Markolie** [comment]
Ah, thanks for the heads up. I'll port it now.

File: code/modules/admin/create_poll.dm (hunk 1 above)
Comment (Jordie0608) [inline code/modules/admin/create_poll.dm]: The magic is dead you monster.

Comment (duncathan) [inline code/modules/admin/create_poll.dm]: <3

**Pull request merged**